airbyte-cdk/python/airbyte_cdk/sources/declarative/transformations/add_fields.py

```diff
@@ -0,0 +1,96 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+from dataclasses import dataclass
+from typing import Any, List, Mapping
+
+import dpath.util
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.transformations import RecordTransformation
+from airbyte_cdk.sources.declarative.types import Config, FieldPointer, StreamSlice, StreamState
+
+
+@dataclass(frozen=True)
+class AddedFieldDefinition:
+    path: FieldPointer
+    value: str
```

Comment:
I think this has to be `Union[InterpolatedString, str] because `InterpolatedString` has an optional `default` field.

the AddFields constructor could either be updated or AddFields could directly process `AddFieldDefinition`

```
            if not isinstance(field.value, InterpolatedString):
                if not isinstance(field.value, str):
                    raise f"Expected a string value for the AddFields transformation: {field}"
                else:
                    self._fields.append(ParsedAddFieldDefinition(field.path, InterpolatedString(field.value)))
            else:
                self._fields.append(ParsedAddFieldDefinition(field.path, field.value))
```

Alternatively, we could do the conversion in the `__post_init__`, but I don't think this will play well with the JsonSchema because it'll complain if the value is a normal string

```
@dataclass(frozen=True)
class AddedFieldDefinition:
    path: FieldPointer
    value: InterpolatedString

    def __post_init__(self):
        if isinstance(self.value, str):
            object.__setattr__(self, "value", InterpolatedString(self.value))
```

Reply:
done